### PR TITLE
Add suppress warnings functionality

### LINKED ISSUE #4263: Provide a way to mute warnings that can only be fixed with access to configuration
Create React App doesn't expose Webpack configuration to the users, so any Webpack warnings that require fixes in config are non-actionable for our users. Unless they are critical (but in this case they should be errors), I would like to have an option to mute them.

https://github.com/webpack/webpack/issues/1617 is a good example of such warning. Create React App users have no control over it (when caused by third party libraries shipping a pre-built file). [Just adding a `noParse` option](https://github.com/webpack/webpack/issues/1617#issuecomment-274342449) wouldn't work for them because they can't change the config. For this warning specifically, there are also some users who consciously choose to use precompiled files (e.g. for build speed), so I'm not sure it's entirely relevant in all cases.

Another example of this being inconvenient is that this warning causes CI to fail for our users. We use `eslint-loader` and due to popular demand configured CI script to treat ESLint warnings as errors on CI. Unfortunately this also means unactionable Webpack warnings break the build. (https://github.com/facebookincubator/create-react-app/issues/1150)

Would you consider a feature request of muting warnings that are only actionable for people with access to the config?

### PR DESCRIPTION
**Pull request by timse**
**What kind of change does this PR introduce?**

feature

**Did you add tests for your changes?**

Yes

**If relevant, link to documentation update:**

TBD

**Summary**

Allows to filter which warnings are to be shown.

**Does this PR introduce a breaking change?**

No

**Other information**

fixes #4263

### CODE REVIEW

**Comment by timse** [comment]
cc @gaearon I hope this helps

**Comment by sokra** [comment]
This only suppress warnings in the output. Warnings should be filtered in the status object not in the toString method. The stats object should only contain JSON-stringable objects.

**Comment by timse** [comment]
anything left to do here?

File: test/statsCases/filter-warnings/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		sourceMap: true,
+		compress: {
+			warnings: true,
+		},
+		mangle: false,
+		beautify: true,
+		comments: false
+	})],
+	stats: {
+		chunkModules: false,
```
Comment (sokra) [inline test/statsCases/filter-warnings/webpack.config.js]: `modules: false` `hash: false` `assets: false` to make the stats smaller

This will cause less changes when unrelated code is updated

Comment (timse) [inline test/statsCases/filter-warnings/webpack.config.js]: changed it :)

**Comment by sokra** [comment]
Thanks

**Comment by corysimmons** [comment]
Is this just implicit now or exposed via some sort of config? If the latter, does the README need updated?

**Comment by timse** [comment]
good that you remind me, this needs an update in the documentation :)

**Comment by corysimmons** [comment]
Thank you timse ^.~

**Comment by timse** [comment]
here is a draft: https://github.com/webpack/webpack.js.org/pull/1092
but you need to do this yourself via the `stats` webpack config options
